admin-dev/themes/new-theme/js/pages/product/components/catalog-price-rule/catalog-price-rule-list.ts

```diff
@@ -0,0 +1,104 @@
+
+  listContainer: HTMLElement
+
+
+  constructor(
+  ) {
+    this.listContainer = document.querySelector(CatalogPriceRuleMap.listContainer) as HTMLElement;
+    this.eventEmitter = window.prestashop.instance.eventEmitter;
+    this.router = new Router();
+  }
+
+  public renderList(): void {
+    const {listFields} = CatalogPriceRuleMap;
+    const tbody = this.listContainer.querySelector(`${CatalogPriceRuleMap.listContainer} tbody`) as HTMLElement;
+    const trTemplateContainer = this.listContainer.querySelector(CatalogPriceRuleMap.listRowTemplate) as HTMLScriptElement;
+    const trTemplate = trTemplateContainer.innerHTML as string;
+    tbody.innerHTML = '';
+
+    getCatalogRules().then((response) => {
+      const catalogPriceRules = response.catalogPriceRules as Array<CatalogPriceRuleForListing>;
+      this.toggleListVisibility(catalogPriceRules.length > 0);
+
+      catalogPriceRules.forEach((catalogPriceRule: CatalogPriceRuleForListing) => {
+        const temporaryContainer = document.createElement('tbody');
+        temporaryContainer.innerHTML = trTemplate.trim();
+
+        const trClone = temporaryContainer.firstChild as HTMLElement;
+        const idField = this.selectListField(trClone, listFields.catalogPriceRuleId);
+        const shopField = this.selectListField(trClone, listFields.shop);
+        const currencyField = this.selectListField(trClone, listFields.currency);
+        const countryField = this.selectListField(trClone, listFields.country);
+        const groupField = this.selectListField(trClone, listFields.group);
+        const nameField = this.selectListField(trClone, listFields.name);
+        const fromQuantityField = this.selectListField(trClone, listFields.fromQuantity);
+        const reductionTypeField = this.selectListField(trClone, listFields.reductionType);
+        const reductionField = this.selectListField(trClone, listFields.reduction);
+        const startDateField = this.selectListField(trClone, listFields.from);
+        const endDateField = this.selectListField(trClone, listFields.to);
+
+        const editBtn = this.selectLink(trClone, listFields.editBtn);
+        idField.textContent = String(catalogPriceRule.id);
+        shopField.textContent = catalogPriceRule.shop;
+        currencyField.textContent = catalogPriceRule.currency;
+        countryField.textContent = catalogPriceRule.country;
+        groupField.textContent = catalogPriceRule.group;
+        nameField.textContent = catalogPriceRule.name;
+        fromQuantityField.textContent = catalogPriceRule.fromQuantity;
+        reductionTypeField.textContent = catalogPriceRule.reductionType;
+        reductionField.textContent = catalogPriceRule.reduction;
+        startDateField.textContent = catalogPriceRule.startDate;
+        endDateField.textContent = catalogPriceRule.endDate;
+        editBtn.href = this.router.generate('admin_catalog_price_rules_edit', {catalogPriceRuleId: String(catalogPriceRule.id)});
```

Comment:
You cannot use this route here, because this page hasn't been officially migrated yet Although the Symfony page is available the `_legacy_link` is still commented meaning the page used in the BO is still the legacy one so you need to redirect to this legacy one